ecl/hql/hqlexpr.hpp

```diff
@@ -1201,6 +1209,37 @@ interface IHqlNamedAnnotation : public IHqlAnnotation
     virtual int getEndPos() const = 0;
 };
 
+#define MAX_SYMBOL_DISTANCE 3
+class NearestSymbol : public CInterface
+{
+public :
+    enum metricType { optimalStringAlignment, damerauLevenshtein, weightedDamerauLevenshtein };
+
+    NearestSymbol(IIdAtom * _searchName, metricType _metric) : distance(-1), metric(_metric)
+    {
+        searchName.set(_searchName);
+    }
+    NearestSymbol(IIdAtom * _searchName) : NearestSymbol(_searchName, weightedDamerauLevenshtein) {}
+    ~NearestSymbol() { searchName.clear(); nearest.clear(); }
+    void reset();
+
+    unsigned computeEditDistance(const IIdAtom * comparison) const;
+    const char * composeSymbolSuggestionMsg(StringBuffer & msg) const;
+    void compare(IHqlExpression * expr);
+    void compare(NearestSymbol & comparison);
+    void compare(NearestSymbol & comparison, unsigned & penalty);
+    bool bestMatchFound() const;
+    IHqlExpression * branchSearch(IHqlSimpleScope * scope, unsigned & penalty);
+    IHqlExpression * branchSearch(IHqlScope * scope, unsigned lookupFlags, HqlLookupContext & ctx, unsigned & penalty);
+    IHqlExpression * querySymbol() const { return nearest; }
+
+private :
+    Owned<IIdAtom> searchName;
+    Owned<IHqlExpression> nearest;
```

Comment:
You should probably be saving the id of the nearest.  Long term (e.g., when comparing against reserved words) it will need to be.